src/telemetry/stats.c

```diff
@@ -0,0 +1,579 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+#include <postgres.h>
+#include <access/table.h>
+#include <access/genam.h>
+#include <access/tableam.h>
+#include <access/htup_details.h>
+#include <catalog/pg_class.h>
+#include <catalog/indexing.h>
+#include <catalog/namespace.h>
+#include <catalog/pg_namespace.h>
+#include <utils/builtins.h>
+#include <fmgr.h>
+
+#include "stats.h"
+#include "ts_catalog/catalog.h"
+#include "ts_catalog/continuous_agg.h"
+#include "chunk.h"
+#include "extension.h"
+#include "hypertable_cache.h"
+#include "utils.h"
+
+typedef struct StatsContext
+{
+	TelemetryStats *stats;
+	ScanIterator compressed_chunk_stats_iterator;
+	bool iterator_valid;
+} StatsContext;
+
+/*
+ * Determine the type of a hypertable.
+ */
+static StatsRelType
+classify_hypertable(const Hypertable *ht)
```

Comment:
HyperTable type is used in this function. `ts_hypertable_get_type()`  doesn't work in this case because it only returns a subset of the types we use here. We might be able to unify this in the future into something more general.